### ISSUE TITLE
Veterancy

### ISSUE THREAD

**Issue by zrisher** [stackcollision/GardenConquest#64]
It has been suggested that we implement some sort of Veterancy system where ships that have survived a number of battles or made a certain number of kills have increased firepower, armor, etc. I think this would be incredibly hard to implement and track because
- there's no sure-fire way for us to identify entities across loads
- how do you determine if a ship has been damaged "too much" for its veterancy to go away? There are so many ways grids can change
- making the upgraded ability work would be complex, because we'd need to adjust the ammos of the weapons, but without doing it for other weapons of that type, and we'd have to do it in tiers....
- upgraded armor would be very complex as well because damage is entirely based on number of components in there, and that's dependent on block definition

**Comment by zrisher** [comment]
This is now significantly more possible with:
https://github.com/KeenSoftwareHouse/SpaceEngineers/pull/314

We could even keep scoreboards of damage done by various types, i.e.:
`The Grinder: Player XYZ - 39840 damage done with a grinder`

**Comment by MeusRex** [comment]
I don't think that this is a good idea. If a ship's design or its pilot is so good that the ship scores over average kills, then there is absolutely no need to make it even stronger...

Also, increasing the damage of a weapon due to the amount of damage it dealt is a very strange concept. If anything the pilot using the weapon should get a bonus, but even that is rather illogical.

**Comment by zrisher** [comment]
> I don't think that this is a good idea. If a ship's design or its pilot is so good that the ship scores over average kills, then there is absolutely no need to make it even stronger...

If this were absolutely true, there would be no need for a stats-boosting veterancy system in any game ever. We've seen a ton of them, so clearly there's something there.

> Also, increasing the damage of a weapon due to the amount of damage it dealt is a very strange concept. If anything the pilot using the weapon should get a bonus, but even that is rather illogical.

We could implement this based on the pilot, damage done by a player is passed with their PlayerID.

There's really no suggestion of how we would fully implement here, this is just a place to start thinking about it. We've had plenty of requests for this from players on the server and there are numerous examples of similar systems in blockbuster games, so I don't think it can be ruled a bad idea outright. Difficult to do well yes, and possibly not worth the time in the short run. But it runs along the lines of this mod's core focus, which is to give more objectives and diverse combat-oriented gameplay experiences to users.
